### PR TITLE
Internal: Import-export CPT UI component [ED-6163]

### PR DESCRIPTION
**Pull request by davseve**

### CODE REVIEW

File: core/app/assets/js/ui/atoms/select.js [hunk 1]
```diff
@@ -19,11 +19,11 @@ export default function Select( props ) {
 }
 Select.propTypes = {
 	className: PropTypes.string,
-	value: PropTypes.string,
 	onChange: PropTypes.func,
 	options: PropTypes.array,
 	elRef: PropTypes.object,
 	multiple: PropTypes.bool,
+	value: PropTypes.oneOfType( [ PropTypes.array, PropTypes.string ] ),
```
Comment (matipojo) [inline core/app/assets/js/ui/atoms/select.js:26]: `<option key={ option.value } value={ option.value }>`
How can it be an array?

Comment (davseve) [reply core/app/assets/js/ui/atoms/select.js:26]: here is the documentation I used to for line 26
https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes

```
  optionalUnion: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.instanceOf(Message)
  ]),
```

File: core/app/modules/import-export/directories/root.php [hunk 2]
```diff
@@ -109,9 +109,21 @@ protected function get_default_sub_directories() {
...
 			$sub_directories[] = new Content( $this->iterator, $this );
 
 			$sub_directories[] = new WP_Content( $this->iterator, $this );
+
+			if ( ! empty( $custom_post_types ) ) {
+				$sub_directories[] = new Custom_Post_Type_Title( $this->iterator, $this );
+			}
+		}
+
+		if ( in_array( 'plugins', $include, true ) ) {
```
Comment (yossi-elementor) [inline core/app/modules/import-export/directories/root.php:125]: Remove lol

**Review by yossi-elementor — CHANGES_REQUESTED** [review]

File: core/app/assets/js/ui/molecules/select2.js [hunk 3]
```diff
@@ -7,9 +7,10 @@ import './select2.scss';
  *
  * @returns {{placeholder: string, allowClear: boolean, dir: string}}
  */
-const getDefaultSettings = () => ( {
+
+const getDefaultSettings = ( placeholder ) => ( {
 	allowClear: true,
-	placeholder: '',
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/assets/js/ui/molecules/select2.js:12]: Resolved

File: core/app/assets/js/ui/molecules/select2.js [hunk 4]
```diff
@@ -27,7 +28,7 @@ export default function Select2( props ) {
 	React.useEffect( () => {
 		const $select2 = jQuery( ref.current )
 			.select2( {
-				...getDefaultSettings(),
+				...getDefaultSettings( props.placeholder ),
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/assets/js/ui/molecules/select2.js:31]: resolved

File: core/app/assets/js/ui/molecules/select2.js [hunk 5]
```diff
@@ -46,18 +47,29 @@ export default function Select2( props ) {
...
+		value={ props.value }
+		onChange={ props.onChange }
+		elRef={ ref }
+		options={ props.options }
+		placeholder={props.placeholder}
+	/>;
 }
 Select2.propTypes = {
-	value: PropTypes.string,
+	value: PropTypes.array,
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/assets/js/ui/molecules/select2.js:61]: Resolved

File: core/app/modules/import-export/assets/js/shared/cpt-select-box/cpt-select-box.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	const sharedContext = useContext( SharedContext ),
+	{ customPostTypes } = sharedContext.data || [],
+	[ selected, setSelected ] = useState( [] );
+
+	const selectedCpt = ( e ) => {
+		setSelected( Array.from( e.target.selectedOptions ).map( ( { value } ) => value ) );
+	};
+
+	useEffect( () => {
+		setSelected( customPostTypes.map( ( item ) => item.value ) );
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/shared/cpt-select-box/cpt-select-box.js:17]: Done, please check that the refactor meet your recommendations :)

File: core/app/modules/import-export/assets/js/shared/cpt-select-box/set-cpt-select-box-existing-options.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export const setCptSelectBoxExistingOptions = ( customPostTypeFromServer, label ) => {
+	const cptOptionsArray = [];
+	// eslint-disable-next-line no-unused-expressions
+	{ customPostTypeFromServer &&
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/shared/cpt-select-box/set-cpt-select-box-existing-options.js:4]: Replaced to `if `condition statement.
Added default value to `label` variable

File: core/app/modules/import-export/directories/custom-post-type-title.php [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+		return 'custom-post-type-title';
+	}
+
+	/**
+	 * @since 3.6.1
+	 *
+	 * @return array
+	 */
+	protected function get_default_sub_directories() {
+		$export_settings = json_decode( stripslashes( $_POST['data'] ), true );
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/directories/custom-post-type-title.php:29]: After seeing how plugins is built it all became clear :)

**Pull request closed without merging**